src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2bail/Cas2BailStatusUpdateService.kt

```diff
@@ -0,0 +1,182 @@
+  @Value("\${url-templates.frontend.cas2.application-overview}") private val applicationOverviewUrlTemplate: String,
+) {
+
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  fun isValidStatus(statusUpdate: Cas2AssessmentStatusUpdate): Boolean {
+    return findActiveStatusByName(statusUpdate.newStatus) != null
+  }
+
+  @Transactional
+  @SuppressWarnings("ReturnCount")
+  fun createForAssessment(
+    assessmentId: UUID,
+    statusUpdate: Cas2AssessmentStatusUpdate,
+    assessor: ExternalUserEntity,
+  ): CasResult<Cas2BailStatusUpdateEntity> {
+    val assessment = cas2BailAssessmentRepository.findByIdOrNull(assessmentId)
+      ?: return CasResult.NotFound()
+
+    val status = findActiveStatusByName(statusUpdate.newStatus)
+      ?: return CasResult.GeneralValidationError("The status ${statusUpdate.newStatus} is not valid")
+
+
+    val newDetails = statusUpdate.newStatusDetails.isNullOrEmpty()
+    val statusDetails = if (newDetails) {
+      emptyList()
+    } else {
+      statusUpdate.newStatusDetails?.map { detail ->
+          status.findStatusDetailOnStatus(detail)
+            ?: return CasResult.GeneralValidationError("The status detail $detail is not valid")
+      }
+    }
+
+    if (ValidationErrors().any()) {
+      return CasResult.FieldValidationError(ValidationErrors())
+    }
+
+    val createdStatusUpdate = cas2BailStatusUpdateRepository.save(
+      Cas2BailStatusUpdateEntity(
+        id = UUID.randomUUID(),
+        assessment = assessment,
+        application = assessment.application,
+        assessor = assessor,
+        statusId = status.id,
+        description = status.description,
+        label = status.label,
+        createdAt = OffsetDateTime.now(),
+      ),
+    )
+
+    statusDetails?.forEach { detail ->
+        cas2BailStatusUpdateDetailRepository.save(
+          Cas2BailStatusUpdateDetailEntity(
+            id = UUID.randomUUID(),
+            statusDetailId = detail.id,
+            statusUpdate = createdStatusUpdate,
+            label = detail.label,
+          ),
+        )
+    }
+
+    sendEmailStatusUpdated(assessment.application.createdByUser, assessment.application, createdStatusUpdate)
+
+    createStatusUpdatedDomainEvent(createdStatusUpdate, statusDetails)
+
+
+    return CasResult.Success(createdStatusUpdate)
+  }
+
+  private fun findActiveStatusByName(statusName: String): Cas2PersistedApplicationStatus? {
+    return statusFinder.active()
+      .find { status -> status.name == statusName }
+  }
+
+  fun createStatusUpdatedDomainEvent(
+    statusUpdate: Cas2BailStatusUpdateEntity,
+    statusDetails: List<Cas2PersistedApplicationStatusDetail>? = emptyList()) {
+    val domainEventId = UUID.randomUUID()
+    val eventOccurredAt = statusUpdate.createdAt
+    val application = statusUpdate.application
+    val newStatus = statusUpdate.status()
+    val assessor = statusUpdate.assessor
+
+    domainEventService.saveCas2ApplicationStatusUpdatedDomainEvent(
```

Comment:
sanity check - are we ok with reusing existing CAS2 domain events for CAS2 bail applications?

Reply:
For now we are using them but we expect them to possibly change.